util/network-devp2p/src/discovery.rs

```diff
@@ -233,17 +249,17 @@ impl Discovery {
 		self.discovery_round += 1;
 	}
 
-	fn distance(a: &H256, b: &H256) -> u32 {
-		let d = *a ^ *b;
-		let mut ret:u32 = 0;
-		for i in 0..32 {
-			let mut v: u8 = d[i];
-			while v != 0 {
-				v >>= 1;
-				ret += 1;
+	/// The base 2 log of the distance between a and b using the XOR metric.
+	fn distance(a: &H256, b: &H256) -> Option<usize> {
+		for i in (0..ADDRESS_BYTES_SIZE).rev() {
+			let byte_index = ADDRESS_BYTES_SIZE - i - 1;
+			let d: u8 = a[byte_index] ^ b[byte_index];
+			if d != 0 {
+				let high_bit_index = FastLog::log2(d);
```

Comment:
Given that we're only using `FastLog::log2` for `u8`, and we've already had some fixed constant there, what about using something like go-ethereum's [lzcount](https://github.com/ethereum/go-ethereum/blob/master/p2p/discover/node.go#L364)? Not sure about the performance impact though.

Reply:
I could change it if you want, I think the performance difference is pretty negligible. Integer log is a generally useful operation, regardless, but on 1-byte values a lookup table works as well I suppose.